tests/components/openai_conversation/test_init.py

```diff
@@ -0,0 +1,68 @@
+"""Tests for the OpenAI integration."""
+
+from homeassistant.components.openai_conversation import OpenAIAgent
+from homeassistant.helpers import device_registry
+
+
+async def test_default_prompt(hass):
+    """Test that the default prompt works."""
+    agent = OpenAIAgent(hass, None)
```

Comment:
Can we test the agent of the integration via the conversion api instead so we don't need to access integration details?

Reply:
this is not to test the agent. This is just to test the prompt generation.

Reply:
Looks like we can use this function:

https://github.com/home-assistant/core/blob/7ddb467ba60c4933deb5aab5fde58076c228030a/homeassistant/components/conversation/__init__.py#L228-L234

to trigger `async_process`.

If we patch the library client we can check that the correct prompt is passed to the library.

Reply:
That's not necessary in this case. These are tests of the integration and so it is the goal to test parts of the integration.

Reply:
I don't agree. We want to test that the correct prompt is sent to the client when we converse with the agent. We should start a conversation and check that the client receives the correct prompt.